### PR TITLE
bluetooth: make beacon samples more usable

### PR DESCRIPTION
**Pull request by pabigot**
As noted on slack the beacon examples currently appear to transmit using an NRPA that can't be retrieved, which makes confirming that the beacons work difficult.  Add an advertising configuration that uses the identity address, which is appropriate for sensor beacons, and display the address that will be used on the console output.

### CODE REVIEW

File: samples/bluetooth/eddystone/src/main.c [hunk 1]
```diff
@@ -426,23 +426,35 @@ static ssize_t read_adv_data(struct bt_conn *conn,
...
+		bt_id_get(&addr, &count);
+		err = bt_le_adv_start(BT_LE_ADV_NCONN_IDENTITY, slot->ad,
 				      ARRAY_SIZE(slot->ad), NULL, 0);
 	}
 
 	if (err) {
 		printk("Advertising failed to start (err %d)\n", err);
 		return err;
 	}
+	bt_addr_le_to_str(&addr, addr_s, sizeof(addr_s));
```
Comment (joerchan) [inline samples/bluetooth/eddystone/src/main.c:456]: blank line after `}`

File: samples/bluetooth/eddystone/src/main.c [hunk 2]
```diff
@@ -426,23 +426,35 @@ static ssize_t read_adv_data(struct bt_conn *conn,
...
 	/* Restart advertising */
 	bt_le_adv_stop();
 
 	if (type == EDS_TYPE_NONE) {
+		struct bt_le_oob oob;
+
 		/* Restore connectable if slot */
+		if (bt_le_oob_get_local(BT_ID_DEFAULT, &oob) == 0) {
+			addr = oob.addr;
+		}
```
Comment (joerchan) [inline samples/bluetooth/eddystone/src/main.c:441]: blank line after `}`

**Comment by joerchan** [comment]
Looks good to me. Just common bluetooth style changes.

**Review by joerchan — APPROVED** [review]

**Comment by pabigot** [comment]
I'd appreciate another review and merge soon, if possible.  I've currently got this within #28891 and would like to rebase that on this if this is merged soon.  If that doesn't happen I'll remove the example refactoring that's based on this PR from that PR, to reduce complexity of that review.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**